combat-trainer.lic

```diff
@@ -1997,7 +2025,186 @@ class TrainerProcess
+      things_to_try = @constellations.select do |data|
+        data['telescope'] &&
+          data['circle'] <= DRStats.circle &&
+          data['circle'] > best_eye_data['circle'] &&
+          (night || !data['constellation']) &&
+          data['pools'].values.compact.size > best_eye_data['pools'].values.compact.size
+      end
+      things_to_try << best_eye_data
+      things_to_try.sort! { |data| data['circle'] }.reverse!
+      echo(" things_to_try = #{things_to_try}") if UserVars.astrology_debug
+      things_to_try.find do |data|
+        result = nil
+        until ['Roundtime', 'You peer aimlessly through your telescope'].include?(result)
+          return check_heavens if Flags['bad-search'] == 'is foiled by the daylight'
+          next if Flags['bad-search'] == 'turns up fruitless'
+          result = observe(data['name'])
+        end
+        result == 'Roundtime'
+      end
+    else
+      until observe(best_eye_data['name'])
+        if Flags['bad-search']
+          check_heavens
+          return
+        end
+      end
+    end
+    pause 2
+    waitrt?
+  end  
+  
+  def observe(body)
+    Flags.add('bad-search', 'is foiled by the (daylight|darkness)', 'turns up fruitless')
+    if @have_telescope
+      DRC.bput('get telescope', 'you get', 'You are already', "That can't be picked up")
+      case DRC.bput("center telescope on #{body}", 'You put your eye', 'open it to make any use of it', 'The pain is too much')
+      when /open it/
+        DRC.bput('open my telescope', 'extend your telescope')
+        return observe(body)
+      when /The pain is too much/
+        return
+        return observe(body)
+      end
+      waitrt?
+      pause 1
+      fput("retreat")
+      fput("retreat")
+      result = DRC.bput('peer telescope', 'You peer aimlessly through your telescope', 'open it to make any use of it', 'You see nothing regarding the future', 'The pain is too much', 'Roundtime')
+      case result
+      when /The pain is too much/
+        return
+        return observe(body)
+      when /open it/
+        DRC.bput('open my telescope', 'extend your telescope')
+        return observe(body)
+      end
+      @equipment_manager.empty_hands
+      waitrt?
+      pause 1   
+      predict_all(check_pools)
+      result
+    else
+      ['Roundtime', 'Clouds obscure'].include? DRC.bput("observe #{body} in sky", 'Your search for', 'You see nothing regarding the future', 'Clouds obscure', 'Roundtime')
+    end
+  end 
 
+  def align(skill)
+    if skill == 'future events'
+      DRC.bput('predict event', 'You focus inwardly')
+      return
+    end
+    DRC.bput("align #{skill}", 'You focus internally')
+    waitrt?
+    if @have_divination_bones
+      roll_bones
+    else
+      waitrt?
+      pause 1
+      fput("retreat")
+      fput("retreat")  
+      DRC.bput('predict future', 'roundtime')
+    end
+    waitrt?
+    pause
+    pause 0.5 while stunned?
+  end  
+  
+  def roll_bones
+    fput("retreat")
```

Comment:
Consider using DRCT.retreat instead of a double fput('retreat')

Reply:
do I only need 1 line with DRCT.retreat?

Reply:
Correct, have a look at https://github.com/rpherbig/dr-scripts/blob/master/common-travel.lic#L75-L82